ddl/index.go

```diff
@@ -602,66 +602,69 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 	return ver, errors.Trace(err)
 }
 
-func onDropIndex(t *meta.Meta, job *model.Job) (ver int64, _ error) {
-	tblInfo, indexInfo, err := checkDropIndex(t, job)
+func onDropIndexes(t *meta.Meta, job *model.Job) (ver int64, _ error) {
+	tblInfo, idxInfos, err := checkDropIndexes(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
+	if len(idxInfos) == 0 {
+		job.State = model.JobStateCancelled
+		return ver, nil
+	}
 
 	dependentHiddenCols := make([]*model.ColumnInfo, 0)
-	for _, indexColumn := range indexInfo.Columns {
-		if tblInfo.Columns[indexColumn.Offset].Hidden {
-			dependentHiddenCols = append(dependentHiddenCols, tblInfo.Columns[indexColumn.Offset])
+	for _, indexInfo := range idxInfos {
+		for _, indexColumn := range indexInfo.Columns {
+			if tblInfo.Columns[indexColumn.Offset].Hidden {
+				dependentHiddenCols = append(dependentHiddenCols, tblInfo.Columns[indexColumn.Offset])
+			}
 		}
 	}
 
-	originalState := indexInfo.State
-	switch indexInfo.State {
+	originalState := idxInfos[0].State
+	switch idxInfos[0].State {
 	case model.StatePublic:
 		// public -> write only
-		indexInfo.State = model.StateWriteOnly
-		if len(dependentHiddenCols) > 0 {
-			firstHiddenOffset := dependentHiddenCols[0].Offset
-			for i := 0; i < len(dependentHiddenCols); i++ {
-				tblInfo.Columns[firstHiddenOffset].State = model.StateWriteOnly
-				// Set this column's offset to the last and reset all following columns' offsets.
-				adjustColumnInfoInDropColumn(tblInfo, firstHiddenOffset)
-			}
-		}
-		ver, err = updateVersionAndTableInfo(t, job, tblInfo, originalState != indexInfo.State)
-		if err != nil {
-			return ver, errors.Trace(err)
-		}
 		job.SchemaState = model.StateWriteOnly
+		setIndicesState(idxInfos, model.StateWriteOnly)
+		setColumnsState(dependentHiddenCols, model.StateWriteOnly)
+		for _, colInfo := range dependentHiddenCols {
+			adjustColumnInfoInDropColumn(tblInfo, colInfo.Offset)
+		}
+		ver, err = updateVersionAndTableInfo(t, job, tblInfo, originalState != idxInfos[0].State)
```

Comment:
change the job state after all these KV actions.
```
if err != nil {
     return ver, errors.Trace(err)
}
job.SchemaState = model.StateWriteOnly
```

Reply:
PTAL

Reply:
It has been modified here